TestAntreaClusterNetworkPolicyStats failing consistently in CI

**Describe the bug**
I don't know what change could have caused this, but it seems that `TestAntreaClusterNetworkPolicyStats` is now failing consistently in Kind tests

Recent job run for main branch:
```
=== RUN   TestAntreaClusterNetworkPolicyStats
Error:     fixtures.go:150: Creating 'antrea-test' K8s Namespace
Error:     fixtures.go:113: Applying Antrea YAML
Error:     fixtures.go:117: Waiting for all Antrea DaemonSet Pods
Error:     fixtures.go:121: Checking CoreDNS deployment
I0611 01:53:07.183077   19703 k8s_util.go:573] Creating/updating ClusterNetworkPolicy cnp1
Error:     antreapolicy_test.go:3048: Got AntreaNetworkPolicy stats: &AntreaClusterNetworkPolicyStats{ObjectMeta:{cnp1    21b6e6a9-7b38-43bf-8223-d2cd5fa1c49a  0 2021-06-11 01:53:07 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:0,Bytes:0,Sessions:0,},RuleTrafficStats:[]RuleTrafficStats{},}
Error:     antreapolicy_test.go:3048: Got AntreaNetworkPolicy stats: &AntreaClusterNetworkPolicyStats{ObjectMeta:{cnp1    21b6e6a9-7b38-43bf-8223-d2cd5fa1c49a  0 2021-06-11 01:53:07 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:0,Bytes:0,Sessions:0,},RuleTrafficStats:[]RuleTrafficStats{},}
Error:     antreapolicy_test.go:3048: Got AntreaNetworkPolicy stats: &AntreaClusterNetworkPolicyStats{ObjectMeta:{cnp1    21b6e6a9-7b38-43bf-8223-d2cd5fa1c49a  0 2021-06-11 01:53:07 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:22,Bytes:1146,Sessions:18,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-allow-97431b6,TrafficStats:TrafficStats{Packets:12,Bytes:676,Sessions:8,},},RuleTrafficStats{Name:ingress-drop-8849a54,TrafficStats:TrafficStats{Packets:10,Bytes:470,Sessions:10,},},},}
Error:     antreapolicy_test.go:3048: Got AntreaNetworkPolicy stats: &AntreaClusterNetworkPolicyStats{ObjectMeta:{cnp1    21b6e6a9-7b38-43bf-8223-d2cd5fa1c49a  0 2021-06-11 01:53:07 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:22,Bytes:1146,Sessions:18,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-allow-97431b6,TrafficStats:TrafficStats{Packets:12,Bytes:676,Sessions:8,},},RuleTrafficStats{Name:ingress-drop-8849a54,TrafficStats:TrafficStats{Packets:10,Bytes:470,Sessions:10,},},},}
Error:     antreapolicy_test.go:3048: Got AntreaNetworkPolicy stats: &AntreaClusterNetworkPolicyStats{ObjectMeta:{cnp1    21b6e6a9-7b38-43bf-8223-d2cd5fa1c49a  0 2021-06-11 01:53:07 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:22,Bytes:1146,Sessions:18,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-allow-97431b6,TrafficStats:TrafficStats{Packets:12,Bytes:676,Sessions:8,},},RuleTrafficStats{Name:ingress-drop-8849a54,TrafficStats:TrafficStats{Packets:10,Bytes:470,Sessions:10,},},},}
Error:     antreapolicy_test.go:3048: Got AntreaNetworkPolicy stats: &AntreaClusterNetworkPolicyStats{ObjectMeta:{cnp1    21b6e6a9-7b38-43bf-8223-d2cd5fa1c49a  0 2021-06-11 01:53:07 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:22,Bytes:1146,Sessions:18,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-allow-97431b6,TrafficStats:TrafficStats{Packets:12,Bytes:676,Sessions:8,},},RuleTrafficStats{Name:ingress-drop-8849a54,TrafficStats:TrafficStats{Packets:10,Bytes:470,Sessions:10,},},},}
Error:     antreapolicy_test.go:3048: Got AntreaNetworkPolicy stats: &AntreaClusterNetworkPolicyStats{ObjectMeta:{cnp1    21b6e6a9-7b38-43bf-8223-d2cd5fa1c49a  0 2021-06-11 01:53:07 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:22,Bytes:1146,Sessions:18,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-allow-97431b6,TrafficStats:TrafficStats{Packets:12,Bytes:676,Sessions:8,},},RuleTrafficStats{Name:ingress-drop-8849a54,TrafficStats:TrafficStats{Packets:10,Bytes:470,Sessions:10,},},},}
Error:     antreapolicy_test.go:3048: Got AntreaNetworkPolicy stats: &AntreaClusterNetworkPolicyStats{ObjectMeta:{cnp1    21b6e6a9-7b38-43bf-8223-d2cd5fa1c49a  0 2021-06-11 01:53:07 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:22,Bytes:1146,Sessions:18,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-allow-97431b6,TrafficStats:TrafficStats{Packets:12,Bytes:676,Sessions:8,},},RuleTrafficStats{Name:ingress-drop-8849a54,TrafficStats:TrafficStats{Packets:10,Bytes:470,Sessions:10,},},},}
Error:     antreapolicy_test.go:3048: Got AntreaNetworkPolicy stats: &AntreaClusterNetworkPolicyStats{ObjectMeta:{cnp1    21b6e6a9-7b38-43bf-8223-d2cd5fa1c49a  0 2021-06-11 01:53:07 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:22,Bytes:1146,Sessions:18,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-allow-97431b6,TrafficStats:TrafficStats{Packets:12,Bytes:676,Sessions:8,},},RuleTrafficStats{Name:ingress-drop-8849a54,TrafficStats:TrafficStats{Packets:10,Bytes:470,Sessions:10,},},},}
Error:     antreapolicy_test.go:3048: Got AntreaNetworkPolicy stats: &AntreaClusterNetworkPolicyStats{ObjectMeta:{cnp1    21b6e6a9-7b38-43bf-8223-d2cd5fa1c49a  0 2021-06-11 01:53:07 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:22,Bytes:1146,Sessions:18,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-allow-97431b6,TrafficStats:TrafficStats{Packets:12,Bytes:676,Sessions:8,},},RuleTrafficStats{Name:ingress-drop-8849a54,TrafficStats:TrafficStats{Packets:10,Bytes:470,Sessions:10,},},},}
Error:     antreapolicy_test.go:3048: Got AntreaNetworkPolicy stats: &AntreaClusterNetworkPolicyStats{ObjectMeta:{cnp1    21b6e6a9-7b38-43bf-8223-d2cd5fa1c49a  0 2021-06-11 01:53:07 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:22,Bytes:1146,Sessions:18,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-allow-97431b6,TrafficStats:TrafficStats{Packets:12,Bytes:676,Sessions:8,},},RuleTrafficStats{Name:ingress-drop-8849a54,TrafficStats:TrafficStats{Packets:10,Bytes:470,Sessions:10,},},},}
Error:     antreapolicy_test.go:3048: Got AntreaNetworkPolicy stats: &AntreaClusterNetworkPolicyStats{ObjectMeta:{cnp1    21b6e6a9-7b38-43bf-8223-d2cd5fa1c49a  0 2021-06-11 01:53:07 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:22,Bytes:1146,Sessions:18,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-allow-97431b6,TrafficStats:TrafficStats{Packets:12,Bytes:676,Sessions:8,},},RuleTrafficStats{Name:ingress-drop-8849a54,TrafficStats:TrafficStats{Packets:10,Bytes:470,Sessions:10,},},},}
Error:     antreapolicy_test.go:3048: Got AntreaNetworkPolicy stats: &AntreaClusterNetworkPolicyStats{ObjectMeta:{cnp1    21b6e6a9-7b38-43bf-8223-d2cd5fa1c49a  0 2021-06-11 01:53:07 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:22,Bytes:1146,Sessions:18,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-allow-97431b6,TrafficStats:TrafficStats{Packets:12,Bytes:676,Sessions:8,},},RuleTrafficStats{Name:ingress-drop-8849a54,TrafficStats:TrafficStats{Packets:10,Bytes:470,Sessions:10,},},},}
Error:     antreapolicy_test.go:3048: Got AntreaNetworkPolicy stats: &AntreaClusterNetworkPolicyStats{ObjectMeta:{cnp1    21b6e6a9-7b38-43bf-8223-d2cd5fa1c49a  0 2021-06-11 01:53:07 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:22,Bytes:1146,Sessions:18,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-allow-97431b6,TrafficStats:TrafficStats{Packets:12,Bytes:676,Sessions:8,},},RuleTrafficStats{Name:ingress-drop-8849a54,TrafficStats:TrafficStats{Packets:10,Bytes:470,Sessions:10,},},},}
Error:     antreapolicy_test.go:3048: Got AntreaNetworkPolicy stats: &AntreaClusterNetworkPolicyStats{ObjectMeta:{cnp1    21b6e6a9-7b38-43bf-8223-d2cd5fa1c49a  0 2021-06-11 01:53:07 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:22,Bytes:1146,Sessions:18,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-allow-97431b6,TrafficStats:TrafficStats{Packets:12,Bytes:676,Sessions:8,},},RuleTrafficStats{Name:ingress-drop-8849a54,TrafficStats:TrafficStats{Packets:10,Bytes:470,Sessions:10,},},},}
Error:     antreapolicy_test.go:3048: Got AntreaNetworkPolicy stats: &AntreaClusterNetworkPolicyStats{ObjectMeta:{cnp1    21b6e6a9-7b38-43bf-8223-d2cd5fa1c49a  0 2021-06-11 01:53:07 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:22,Bytes:1146,Sessions:18,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-allow-97431b6,TrafficStats:TrafficStats{Packets:12,Bytes:676,Sessions:8,},},RuleTrafficStats{Name:ingress-drop-8849a54,TrafficStats:TrafficStats{Packets:10,Bytes:470,Sessions:10,},},},}
Error:     antreapolicy_test.go:3048: Got AntreaNetworkPolicy stats: &AntreaClusterNetworkPolicyStats{ObjectMeta:{cnp1    21b6e6a9-7b38-43bf-8223-d2cd5fa1c49a  0 2021-06-11 01:53:07 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:22,Bytes:1146,Sessions:18,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-allow-97431b6,TrafficStats:TrafficStats{Packets:12,Bytes:676,Sessions:8,},},RuleTrafficStats{Name:ingress-drop-8849a54,TrafficStats:TrafficStats{Packets:10,Bytes:470,Sessions:10,},},},}
Error:     antreapolicy_test.go:3048: Got AntreaNetworkPolicy stats: &AntreaClusterNetworkPolicyStats{ObjectMeta:{cnp1    21b6e6a9-7b38-43bf-8223-d2cd5fa1c49a  0 2021-06-11 01:53:07 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:22,Bytes:1146,Sessions:18,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-allow-97431b6,TrafficStats:TrafficStats{Packets:12,Bytes:676,Sessions:8,},},RuleTrafficStats{Name:ingress-drop-8849a54,TrafficStats:TrafficStats{Packets:10,Bytes:470,Sessions:10,},},},}
Error:     antreapolicy_test.go:3048: Got AntreaNetworkPolicy stats: &AntreaClusterNetworkPolicyStats{ObjectMeta:{cnp1    21b6e6a9-7b38-43bf-8223-d2cd5fa1c49a  0 2021-06-11 01:53:07 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:22,Bytes:1146,Sessions:18,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-allow-97431b6,TrafficStats:TrafficStats{Packets:12,Bytes:676,Sessions:8,},},RuleTrafficStats{Name:ingress-drop-8849a54,TrafficStats:TrafficStats{Packets:10,Bytes:470,Sessions:10,},},},}
E0611 01:54:55.292467   19703 antreapolicy_test.go:78] timed out waiting for the condition
I0611 01:54:55.295805   19703 k8s_util.go:602] Deleting AntreaClusterNetworkPolicies cnp1
I0611 01:54:55.306646   19703 k8s_util.go:811] Deleting test Namespace x
E0611 01:54:55.309418   19703 k8s_util.go:813] Error when deleting Namespace 'x': namespaces "x" not found
I0611 01:54:55.309457   19703 k8s_util.go:811] Deleting test Namespace y
E0611 01:54:55.312474   19703 k8s_util.go:813] Error when deleting Namespace 'y': namespaces "y" not found
I0611 01:54:55.312508   19703 k8s_util.go:811] Deleting test Namespace z
E0611 01:54:55.314796   19703 k8s_util.go:813] Error when deleting Namespace 'z': namespaces "z" not found
Error:     antreapolicy_test.go:80: test failed: timed out waiting for the condition
Error:     fixtures.go:356: Deleting Pod 'test-client-fdedx21r'
Error:     fixtures.go:356: Deleting Pod 'test-server-n7lekh0e'
Error:     fixtures.go:224: Exporting test logs to '/home/runner/work/antrea/antrea/log/TestAntreaClusterNetworkPolicyStats/beforeTeardown.Jun11-01-54-55'
Error:     fixtures.go:349: Deleting 'antrea-test' K8s Namespace
--- FAIL: TestAntreaClusterNetworkPolicyStats (160.80s)
```

It has been failing consistently for 2 of my PRs as well.

I see `TestANPNetworkPolicyStatsWithDropAction` failing a lot as well, so maybe a general issue with policy stats:
```
=== RUN   TestANPNetworkPolicyStatsWithDropAction
Error:     fixtures.go:150: Creating 'antrea-test' K8s Namespace
Error:     fixtures.go:113: Applying Antrea YAML
Error:     fixtures.go:117: Waiting for all Antrea DaemonSet Pods
Error:     fixtures.go:121: Checking CoreDNS deployment
I0611 01:55:45.427116   19871 k8s_util.go:626] Creating/updating Antrea NetworkPolicy antrea-test/np1
Error:     antreapolicy_test.go:2896: Got AntreaNetworkPolicy stats: &AntreaNetworkPolicyStats{ObjectMeta:{np1  antrea-test  0be74435-d868-468c-8cc0-639a13096954  0 2021-06-11 01:55:45 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:0,Bytes:0,Sessions:0,},RuleTrafficStats:[]RuleTrafficStats{},}
Error:     antreapolicy_test.go:2896: Got AntreaNetworkPolicy stats: &AntreaNetworkPolicyStats{ObjectMeta:{np1  antrea-test  0be74435-d868-468c-8cc0-639a13096954  0 2021-06-11 01:55:45 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:0,Bytes:0,Sessions:0,},RuleTrafficStats:[]RuleTrafficStats{},}
Error:     antreapolicy_test.go:2896: Got AntreaNetworkPolicy stats: &AntreaNetworkPolicyStats{ObjectMeta:{np1  antrea-test  0be74435-d868-468c-8cc0-639a13096954  0 2021-06-11 01:55:45 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:0,Bytes:0,Sessions:0,},RuleTrafficStats:[]RuleTrafficStats{},}
Error:     antreapolicy_test.go:2896: Got AntreaNetworkPolicy stats: &AntreaNetworkPolicyStats{ObjectMeta:{np1  antrea-test  0be74435-d868-468c-8cc0-639a13096954  0 2021-06-11 01:55:45 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:0,Bytes:0,Sessions:0,},RuleTrafficStats:[]RuleTrafficStats{},}
Error:     antreapolicy_test.go:2896: Got AntreaNetworkPolicy stats: &AntreaNetworkPolicyStats{ObjectMeta:{np1  antrea-test  0be74435-d868-468c-8cc0-639a13096954  0 2021-06-11 01:55:45 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:24,Bytes:1352,Sessions:16,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-drop-e8d8c49,TrafficStats:TrafficStats{Packets:6,Bytes:282,Sessions:6,},},RuleTrafficStats{Name:ingress-allow-088c867,TrafficStats:TrafficStats{Packets:18,Bytes:1070,Sessions:10,},},},}
Error:     antreapolicy_test.go:2896: Got AntreaNetworkPolicy stats: &AntreaNetworkPolicyStats{ObjectMeta:{np1  antrea-test  0be74435-d868-468c-8cc0-639a13096954  0 2021-06-11 01:55:45 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:24,Bytes:1352,Sessions:16,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-drop-e8d8c49,TrafficStats:TrafficStats{Packets:6,Bytes:282,Sessions:6,},},RuleTrafficStats{Name:ingress-allow-088c867,TrafficStats:TrafficStats{Packets:18,Bytes:1070,Sessions:10,},},},}
Error:     antreapolicy_test.go:2896: Got AntreaNetworkPolicy stats: &AntreaNetworkPolicyStats{ObjectMeta:{np1  antrea-test  0be74435-d868-468c-8cc0-639a13096954  0 2021-06-11 01:55:45 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:24,Bytes:1352,Sessions:16,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-drop-e8d8c49,TrafficStats:TrafficStats{Packets:6,Bytes:282,Sessions:6,},},RuleTrafficStats{Name:ingress-allow-088c867,TrafficStats:TrafficStats{Packets:18,Bytes:1070,Sessions:10,},},},}
Error:     antreapolicy_test.go:2896: Got AntreaNetworkPolicy stats: &AntreaNetworkPolicyStats{ObjectMeta:{np1  antrea-test  0be74435-d868-468c-8cc0-639a13096954  0 2021-06-11 01:55:45 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:24,Bytes:1352,Sessions:16,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-drop-e8d8c49,TrafficStats:TrafficStats{Packets:6,Bytes:282,Sessions:6,},},RuleTrafficStats{Name:ingress-allow-088c867,TrafficStats:TrafficStats{Packets:18,Bytes:1070,Sessions:10,},},},}
Error:     antreapolicy_test.go:2896: Got AntreaNetworkPolicy stats: &AntreaNetworkPolicyStats{ObjectMeta:{np1  antrea-test  0be74435-d868-468c-8cc0-639a13096954  0 2021-06-11 01:55:45 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:24,Bytes:1352,Sessions:16,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-drop-e8d8c49,TrafficStats:TrafficStats{Packets:6,Bytes:282,Sessions:6,},},RuleTrafficStats{Name:ingress-allow-088c867,TrafficStats:TrafficStats{Packets:18,Bytes:1070,Sessions:10,},},},}
Error:     antreapolicy_test.go:2896: Got AntreaNetworkPolicy stats: &AntreaNetworkPolicyStats{ObjectMeta:{np1  antrea-test  0be74435-d868-468c-8cc0-639a13096954  0 2021-06-11 01:55:45 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:24,Bytes:1352,Sessions:16,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-drop-e8d8c49,TrafficStats:TrafficStats{Packets:6,Bytes:282,Sessions:6,},},RuleTrafficStats{Name:ingress-allow-088c867,TrafficStats:TrafficStats{Packets:18,Bytes:1070,Sessions:10,},},},}
Error:     antreapolicy_test.go:2896: Got AntreaNetworkPolicy stats: &AntreaNetworkPolicyStats{ObjectMeta:{np1  antrea-test  0be74435-d868-468c-8cc0-639a13096954  0 2021-06-11 01:55:45 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:24,Bytes:1352,Sessions:16,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-drop-e8d8c49,TrafficStats:TrafficStats{Packets:6,Bytes:282,Sessions:6,},},RuleTrafficStats{Name:ingress-allow-088c867,TrafficStats:TrafficStats{Packets:18,Bytes:1070,Sessions:10,},},},}
Error:     antreapolicy_test.go:2896: Got AntreaNetworkPolicy stats: &AntreaNetworkPolicyStats{ObjectMeta:{np1  antrea-test  0be74435-d868-468c-8cc0-639a13096954  0 2021-06-11 01:55:45 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:24,Bytes:1352,Sessions:16,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-drop-e8d8c49,TrafficStats:TrafficStats{Packets:6,Bytes:282,Sessions:6,},},RuleTrafficStats{Name:ingress-allow-088c867,TrafficStats:TrafficStats{Packets:18,Bytes:1070,Sessions:10,},},},}
Error:     antreapolicy_test.go:2896: Got AntreaNetworkPolicy stats: &AntreaNetworkPolicyStats{ObjectMeta:{np1  antrea-test  0be74435-d868-468c-8cc0-639a13096954  0 2021-06-11 01:55:45 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:24,Bytes:1352,Sessions:16,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-drop-e8d8c49,TrafficStats:TrafficStats{Packets:6,Bytes:282,Sessions:6,},},RuleTrafficStats{Name:ingress-allow-088c867,TrafficStats:TrafficStats{Packets:18,Bytes:1070,Sessions:10,},},},}
Error:     antreapolicy_test.go:2896: Got AntreaNetworkPolicy stats: &AntreaNetworkPolicyStats{ObjectMeta:{np1  antrea-test  0be74435-d868-468c-8cc0-639a13096954  0 2021-06-11 01:55:45 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:24,Bytes:1352,Sessions:16,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-drop-e8d8c49,TrafficStats:TrafficStats{Packets:6,Bytes:282,Sessions:6,},},RuleTrafficStats{Name:ingress-allow-088c867,TrafficStats:TrafficStats{Packets:18,Bytes:1070,Sessions:10,},},},}
Error:     antreapolicy_test.go:2896: Got AntreaNetworkPolicy stats: &AntreaNetworkPolicyStats{ObjectMeta:{np1  antrea-test  0be74435-d868-468c-8cc0-639a13096954  0 2021-06-11 01:55:45 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:24,Bytes:1352,Sessions:16,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-drop-e8d8c49,TrafficStats:TrafficStats{Packets:6,Bytes:282,Sessions:6,},},RuleTrafficStats{Name:ingress-allow-088c867,TrafficStats:TrafficStats{Packets:18,Bytes:1070,Sessions:10,},},},}
Error:     antreapolicy_test.go:2896: Got AntreaNetworkPolicy stats: &AntreaNetworkPolicyStats{ObjectMeta:{np1  antrea-test  0be74435-d868-468c-8cc0-639a13096954  0 2021-06-11 01:55:45 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:24,Bytes:1352,Sessions:16,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-drop-e8d8c49,TrafficStats:TrafficStats{Packets:6,Bytes:282,Sessions:6,},},RuleTrafficStats{Name:ingress-allow-088c867,TrafficStats:TrafficStats{Packets:18,Bytes:1070,Sessions:10,},},},}
Error:     antreapolicy_test.go:2896: Got AntreaNetworkPolicy stats: &AntreaNetworkPolicyStats{ObjectMeta:{np1  antrea-test  0be74435-d868-468c-8cc0-639a13096954  0 2021-06-11 01:55:45 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:24,Bytes:1352,Sessions:16,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-drop-e8d8c49,TrafficStats:TrafficStats{Packets:6,Bytes:282,Sessions:6,},},RuleTrafficStats{Name:ingress-allow-088c867,TrafficStats:TrafficStats{Packets:18,Bytes:1070,Sessions:10,},},},}
Error:     antreapolicy_test.go:2896: Got AntreaNetworkPolicy stats: &AntreaNetworkPolicyStats{ObjectMeta:{np1  antrea-test  0be74435-d868-468c-8cc0-639a13096954  0 2021-06-11 01:55:45 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:24,Bytes:1352,Sessions:16,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-drop-e8d8c49,TrafficStats:TrafficStats{Packets:6,Bytes:282,Sessions:6,},},RuleTrafficStats{Name:ingress-allow-088c867,TrafficStats:TrafficStats{Packets:18,Bytes:1070,Sessions:10,},},},}
Error:     antreapolicy_test.go:2896: Got AntreaNetworkPolicy stats: &AntreaNetworkPolicyStats{ObjectMeta:{np1  antrea-test  0be74435-d868-468c-8cc0-639a13096954  0 2021-06-11 01:55:45 +0000 UTC <nil> <nil> map[] map[] [] []  []},TrafficStats:TrafficStats{Packets:24,Bytes:1352,Sessions:16,},RuleTrafficStats:[]RuleTrafficStats{RuleTrafficStats{Name:ingress-drop-e8d8c49,TrafficStats:TrafficStats{Packets:6,Bytes:282,Sessions:6,},},RuleTrafficStats{Name:ingress-allow-088c867,TrafficStats:TrafficStats{Packets:18,Bytes:1070,Sessions:10,},},},}
E0611 01:57:25.657658   19871 antreapolicy_test.go:78] timed out waiting for the condition
I0611 01:57:25.663840   19871 k8s_util.go:811] Deleting test Namespace x
E0611 01:57:25.666220   19871 k8s_util.go:813] Error when deleting Namespace 'x': namespaces "x" not found
I0611 01:57:25.666250   19871 k8s_util.go:811] Deleting test Namespace y
E0611 01:57:25.668222   19871 k8s_util.go:813] Error when deleting Namespace 'y': namespaces "y" not found
I0611 01:57:25.668254   19871 k8s_util.go:811] Deleting test Namespace z
E0611 01:57:25.670222   19871 k8s_util.go:813] Error when deleting Namespace 'z': namespaces "z" not found
Error:     antreapolicy_test.go:80: test failed: timed out waiting for the condition
Error:     fixtures.go:356: Deleting Pod 'test-client-w6ih8vbm'
Error:     fixtures.go:356: Deleting Pod 'test-server-57e93sik'
Error:     fixtures.go:224: Exporting test logs to '/home/runner/work/antrea/antrea/log/TestANPNetworkPolicyStatsWithDropAction/beforeTeardown.Jun11-01-57-25'
Error:     fixtures.go:349: Deleting 'antrea-test' K8s Namespace
--- FAIL: TestANPNetworkPolicyStatsWithDropAction (169.62s)
```

--- a/test/e2e/antreapolicy_test.go
+++ b/test/e2e/antreapolicy_test.go
@@ -55,7 +55,10 @@ var (
 
 const (
 	// Provide enough time for policies to be enforced & deleted by the CNI plugin.
-	networkPolicyDelay = 1500 * time.Millisecond
+	networkPolicyDelay = 2000 * time.Millisecond
+	// Timeout when waiting for a policy status to be updated and for the
+	// policy to be considered realized.
+	policyRealizedTimeout = 5 * time.Second
 	// provide enough time for groups to have members computed.
 	groupDelay = time.Second
 	// Verification of deleting/creating resources timed out.
@@ -2743,7 +2746,7 @@ func TestAntreaPolicyStatus(t *testing.T) {
 		CurrentNodesRealized: 2,
 		DesiredNodesRealized: 2,
 	}
-	err = wait.Poll(100*time.Millisecond, 3*time.Second, func() (bool, error) {
+	err = wait.Poll(100*time.Millisecond, policyRealizedTimeout, func() (bool, error) {
 		anp, err := data.crdClient.CrdV1alpha1().NetworkPolicies(anp.Namespace).Get(context.TODO(), anp.Name, metav1.GetOptions{})
 		if err != nil {
 			return false, err
@@ -2751,7 +2754,7 @@ func TestAntreaPolicyStatus(t *testing.T) {
 		return anp.Status == expectedStatus, nil
 	})
 	assert.NoError(t, err, "Antrea NetworkPolicy failed to reach expected status")
-	err = wait.Poll(100*time.Millisecond, 3*time.Second, func() (bool, error) {
+	err = wait.Poll(100*time.Millisecond, policyRealizedTimeout, func() (bool, error) {
 		anp, err := data.crdClient.CrdV1alpha1().ClusterNetworkPolicies().Get(context.TODO(), acnp.Name, metav1.GetOptions{})
 		if err != nil {
 			return false, err
@@ -2759,6 +2762,40 @@ func TestAntreaPolicyStatus(t *testing.T) {
 		return anp.Status == expectedStatus, nil
 	})
 	assert.NoError(t, err, "Antrea ClusterNetworkPolicy failed to reach expected status")
+}
+
+// waitForANPRealized waits untils an ANP is realized and returns, or times out. A policy is
+// considered realized when its Status has been updated with DesiredNodesRealized > 0 and
+// CurrentNodesRealized == DesiredNodesRealized.
+func (data *TestData) waitForANPRealized(t *testing.T, namespace string, name string) error {
+	t.Logf("Waiting for ANP '%s/%s' to be realized", namespace, name)
+	if err := wait.Poll(100*time.Millisecond, policyRealizedTimeout, func() (bool, error) {
+		anp, err := data.crdClient.CrdV1alpha1().NetworkPolicies(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		return anp.Status.DesiredNodesRealized > 0 && anp.Status.CurrentNodesRealized == anp.Status.DesiredNodesRealized, nil
+	}); err != nil {
+		return fmt.Errorf("error when waiting for ANP '%s/%s' to be realized: %v", namespace, name, err)
+	}
+	return nil
+}
+
+// waitForACNPRealized waits untils an ACNP is realized and returns, or times out. A policy is
+// considered realized when its Status has been updated with DesiredNodesRealized > 0 and
+// CurrentNodesRealized == DesiredNodesRealized.
+func (data *TestData) waitForACNPRealized(t *testing.T, name string) error {
+	t.Logf("Waiting for ACNP '%s' to be realized", name)
+	if err := wait.Poll(100*time.Millisecond, policyRealizedTimeout, func() (bool, error) {
+		acnp, err := data.crdClient.CrdV1alpha1().ClusterNetworkPolicies().Get(context.TODO(), name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		return acnp.Status.DesiredNodesRealized > 0 && acnp.Status.CurrentNodesRealized == acnp.Status.DesiredNodesRealized, nil
+	}); err != nil {
+		return fmt.Errorf("error when waiting for ACNP '%s' to be realized: %v", name, err)
+	}
+	return nil
 }
 
 // TestANPNetworkPolicyStatsWithDropAction tests antreanetworkpolicystats can correctly collect dropped packets stats from ANP if
@@ -2855,8 +2892,8 @@ func TestANPNetworkPolicyStatsWithDropAction(t *testing.T) {
 		failOnError(fmt.Errorf("create ANP failed for ANP %s: %v", anp.Name, err), t)
 	}
 
-	// Wait for a few seconds in case that connections are established before policies are enforced.
-	time.Sleep(networkPolicyDelay)
+	// Wait for the policy to be realized before attempting connections
+	failOnError(data.waitForANPRealized(t, anp.Namespace, anp.Name), t)
 
 	sessionsPerAddressFamily := 10
 	var wg sync.WaitGroup
@@ -3007,8 +3044,8 @@ func TestAntreaClusterNetworkPolicyStats(t *testing.T) {
 		failOnError(fmt.Errorf("create ACNP failed for ACNP %s: %v", acnp.Name, err), t)
 	}
 
-	// Wait for a few seconds in case that connections are established before policies are enforced.
-	time.Sleep(networkPolicyDelay)
+	// Wait for the policy to be realized before attempting connections
+	failOnError(data.waitForACNPRealized(t, acnp.Name), t)
 
 	sessionsPerAddressFamily := 10
 	var wg sync.WaitGroup

